File: src/FASTCAMPUS/ALGO/D35/P15900.js
const fs = require('fs');
class Scanner {
  constructor(fileName){
    this.read = fs.readFileSync(fileName).toString().trim().split('\n');
    this.stringToken = [];
  }
  next(){
    if(this.stringToken.length === 0){
      if(this.read.length > 0){
        this.stringToken = this.read.shift().split(" ");
      } else {
        return null;
      }
    } 
    return this.stringToken.shift();
  }

  nextNumber(){
    const next = this.next();
    if(next){
      return Number(next);
    } else {
      return null;
    }
    
  }

  nextLine(){
    if(this.read.length > 0){
      return this.read.shift();
    } else {
      return null;
    }
    
  }
}
const scanner = new Scanner('stdin.txt');
let N, count;
let graph;
const input = function (){
  count = 0;
  N = scanner.nextNumber();
  graph = Array.from({length : N + 1}, () => new Array());
  scanner.read.forEach(e => {
    //console.log(e.split(" ").map(Number));
    let [a, b] = e.split(" ").map(Number);
    graph[a].push(b);
    graph[b].push(a);
  })
  //console.log(graph);
}

function dfs(s, depth, prev){
  if(s !== 1 && graph[s].length === 1){
    count += depth;
    return;
  }
  graph[s].forEach(e => {
    if(prev === e) return;
    dfs(e, depth + 1, s);
  })
}
const pro = function (){
  dfs(1, 0, 0);
  if(count % 2 === 0){
    console.log('No')
  } else {
    console.log('Yes');
  }
  
}

const solve = function(){
  input();
  pro();
}

solve();

File: src/FASTCAMPUS/ALGO/D35/P15900.java
package FASTCAMPUS.ALGO.D35;
import java.io.*;
import java.util.*;

public class P15900 {
    static int N, count;
    static ArrayList<Integer> [] graph;
    static void input() {
        FastReader scan = new FastReader();
        count = 0;
        N = scan.nextInt();
        graph = new ArrayList[N + 1];
        for(int i = 1; i <= N; i ++){
            graph[i] = new ArrayList<Integer>();
        }
        for(int i = 1; i < N; i ++){
            int a = scan.nextInt();
            int b = scan.nextInt();
            graph[a].add(b);
            graph[b].add(a);
        }
        
    }
    static void dfs(int s, int depth, int prev){
        if(s != 1 && graph[s].size() == 1){
          count += depth;
          return;
        }
        for (Integer e : graph[s]) {
            if(prev == e) continue;  
            dfs(e, depth + 1, s);
        }
      }
    public static void pro(){
        dfs(1, 0, 0);
        if(count % 2 == 0){
            System.out.println("No");
        } else {
            System.out.println("Yes");
        }
    }
    public static void main(String[] args) {
        input();
        pro();
    }
    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        public FastReader(String s) throws FileNotFoundException {
            br = new BufferedReader(new FileReader(new File(s)));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }

        long nextLong() {
            return Long.parseLong(next());
        }

        double nextDouble() {
            return Double.parseDouble(next());
        }

        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}